addon/lib/metrics.js

```diff
@@ -160,34 +137,33 @@ const Metrics = module.exports = {
     const dataParsed = variantMaker.parseTests(JSON.parse(data));
 
     store.experimentVariants[subject] = dataParsed;
-    Metrics.experimentFeaturesChanged(subject, dataParsed);
     Events.emit(EVENT_SEND_VARIANTS, {
       data: JSON.stringify(dataParsed),
       subject: self.id
     });
   },
 
   onExperimentPing: function(ev) {
+    const timestamp = Date.now();
     const { subject, data } = ev;
     const dataParsed = JSON.parse(data);
 
-    // TODO: Map add-on ID (subject) to other pingTypes as necessary
-    const pingType = 'testpilottest';
-
-    if (store.experimentVariants && subject in store.experimentVariants) {
-      dataParsed.variants = store.experimentVariants[subject];
-    }
-
-    const payload = {
-      version: 1,
-      test: subject,
-      payload: dataParsed
-    };
+    AddonManager.getAddonByID(subject, addon => {
```

Comment:
There's a problem here: if this ping happens in an experiment's `unload` handler, this function will return `null` in the lookup. If add-ons are using the event-driven model appropriately this shouldn't be a problem, but Tab Center [works on a schedule](https://github.com/bwinton/TabCenter/blob/1e95d534c4bf37c5a66cb8aa85ea1459cfee6982/main.js#L98) right now, and [forces a flush](https://github.com/bwinton/TabCenter/blob/1e95d534c4bf37c5a66cb8aa85ea1459cfee6982/main.js#L102-L104) on the `unload` handler. This will cause `payload.version` to be null in the latter case here.

CC @clouserw, @bwinton